src/system-plugins/platform-manager/modules/TridentSerialBridge.js

```diff
@@ -0,0 +1,145 @@
+const crc           = require('crc');
+const SerialPort    = require('serialport');
+const Delimiter      = SerialPort.parsers.Delimiter;
+const EventEmitter  = require('events').EventEmitter;
+
+class SerialBridge extends EventEmitter
+{
+  constructor( uartDevicePath, uartBaud )
+  {
+    super();
+
+    this.uartDevicePath = uartDevicePath;
+    this.uartBaud = uartBaud;
+    this.emitRawSerial = false;
+    this.serialConnected = false;
+    this.serialPort = {};
+  }
+
+  connect()
+  {
+    this.serialPort = new SerialPort( this.uartDevicePath, 
+    {
+      baudRate: this.uartBaud,
+      autoOpen: true
+    });
+
+    // All discrete firmware messages end with ';'
+    var parser = this.serialPort.pipe( new Delimiter( { delimiter: ';' } ) );
+
+    this.serialPort.on('open', () =>
+    {
+      this.serialConnected = true;
+      console.log('Serial port opened!');
+    });
+
+    this.serialPort.on('error',(err) =>
+    {
+      console.log('Serial error',err)
+    })
+
+    this.serialPort.on('close', (data) =>
+    {
+      console.log('Serial port closed!');
+      this.serialConnected = false;
+    });
+
+    parser.on('data', (data) =>
+    {
+      var status = this.parseStatus( data.toString('utf8' ) );
+
+      // If valid status message received, emit status events
+      if( status !== null )
+      {
+        this.emit('status', status);
+
+        if( this.emitRawSerial ) 
+        {
+          this.emit('serial-recieved', data + '\n');
+        }
+      }
+    });
+  }
+
+  close()
+  {
+    if( !this.serialConnected )
+    {
+      return;
+    }
+
+    this.serialConnected = false;
+
+    //This code is a work around for a race condition in the serial port code https://github.com/voodootikigod/node-serialport/issues/241#issuecomment-43058353
+    var sp = this.serialPort;
+
+    this.serialPort.flush( (err) =>
+    {
+        this.serialPort.close( (err) => {} );
+    });
+  }
+
+  write( command )
+  {
+    // Create buffer for crc+command
+    let messagebuffer = new Buffer( command.length + 1 );
```

Comment:
Yes, the GC will clean this up after messagebuffer is no longer referenced (after the end of this function/when serialport is done using it)